sample_information_validator.py

```diff
@@ -23,78 +23,280 @@
-    passes = 0
-    total = 0
-    for row_nr in range(row_start, row_end):
-        total += 1
-        cell_id = "{col}{row_nr}".format(col=column_letter, row_nr=row_nr)
-        result = validator.validate(sheet[cell_id])
-        if result:  # Test passed
-            passes += 1
+            logger.warning('Sample concentration ({}ng/ul) in cell {} is out of specifications: {}-{}ng/ul'\
+            .format(self.access_sample_info_sheet[self.concentrationID].value,self.concentrationID, min_conc, max_conc))
+        return True
 
-    logger.info(
-        'Checked column {}. {}/{} passes'.format(column_letter, passes, total)
-        )
+    def validate_vol(self, vol):
+        """Checks entry for volume"""
+        if(self.access_sample_info_sheet[self.volumeID].value < vol):
+            logger.warning('Sample volume ({}ul) in cell {} is lower than required: {}ul'\
+            .format(self.access_sample_info_sheet[self.volumeID].value,self.access_sample_info_sheet[self.volumeID].coordinate, vol))
+            global WARNINGS
+            WARNINGS += 1
+            return True
+        else:
+            return True
+
+    def validate_rin(self):
+        """Checks entry for RIN in RNA samples only"""
+        if self.access_sample_info_sheet[self.rinID].value <8:
+            logger.warning(
+                'RIN value in cell {} is below recommendation'.format(self.access_sample_info_sheet[self.rinID].coordinate)
+            )
+            global WARNINGS
+            WARNINGS += 1
+            return True
+        else:
+            return True
 
 
-def main(input_sheet):
-    wb = load_workbook(input_sheet, read_only=True, data_only=True)
-    ws = wb[SHEET_NAME]
-    nr_samples = 8  # TODO
-    final_row = FIRST_LINE + nr_samples
-    validate_column(ws, CONC_COL, FIRST_LINE, final_row, NumericValidator)
+def main(input_sheet, username_couchDB, password_couchDB, url_couchDB, recom_path):
+    # Instantiate the ProjectSheet object
+    sheetOI = ProjectSheet(input_sheet)
+    # get Project Information from couchDB
+    Project_Information = sheetOI.ProjectInfo(username_couchDB, password_couchDB, url_couchDB)
+    # validate the project name to ensure correct identification in couchDB
+    sheetOI.validate_project_Name(Project_Information)
+    # get info about prep type
+    prep_recommendations = sheetOI.prep_standards(Project_Information, recom_path)
+    # validate all entries
+    sheetOI.validate_column(Project_Information, recom_path)
 
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description=__doc__)
-    parser.add_argument('sample_info_sheet',
+    parser.add_argument('sampleInfoSheet',
                         help="Completed sample info sent to NGI by the user.")
+    parser.add_argument('username', default=None,
+                        help="Username for couchDB")
+    parser.add_argument('password', default=None,
+                        help="Password for couchDB")
```

Comment:
I think we should use a config file with username, password and url. It's just neater! :)